plugins/modules/ec2_instance.py

```diff
@@ -1699,6 +1751,48 @@ def handle_existing(existing_matches, state):
     return result
 
 
+def enforce_count(existing_matches, module):
+    exact_count = module.params.get('exact_count')
+
+    try:
+        current_count = len(existing_matches)
+        if current_count == exact_count:
+            module.exit_json(
+                changed=False,
+                msg='{0} instances already running, nothing to do.'.format(exact_count)
+            )
+
+        elif current_count < exact_count:
+            to_launch = exact_count - current_count
+            module.params['to_launch'] = to_launch
```

Comment:
We could overwrite `module.params['count']` here to avoid the extra code path in `build_run_instance_spec`

@jillr do you have any opinions on this?

Reply:
I generally don't like overwriting module.params, I worry that it's asking for issues down the road when we make some other change that uses `count` and expects it to be whatever the user originally specified.  I think I'd rather have the extra conditional?